test/0dt/mzcompose.py

```diff
@@ -11,24 +11,51 @@
 import time
 from textwrap import dedent
 
+from pg8000.exceptions import InterfaceError
+
 from materialize import buildkite
 from materialize.mzcompose.composition import Composition
+from materialize.mzcompose.services.cockroach import Cockroach
 from materialize.mzcompose.services.kafka import Kafka
 from materialize.mzcompose.services.materialized import Materialized
 from materialize.mzcompose.services.mysql import MySql
 from materialize.mzcompose.services.postgres import Postgres
 from materialize.mzcompose.services.schema_registry import SchemaRegistry
 from materialize.mzcompose.services.testdrive import Testdrive
 from materialize.mzcompose.services.zookeeper import Zookeeper
+from materialize.ui import CommandFailureCausedUIError
+
+DEFAULT_TIMEOUT = "300s"
 
 SERVICES = [
     MySql(),
     Postgres(),
     Zookeeper(),
     Kafka(),
     SchemaRegistry(),
-    Materialized(sanity_restart=False),
-    Testdrive(materialize_params={"cluster": "cluster"}, no_reset=True, seed=1),
+    Cockroach(),
+    Materialized(
+        name="mz_old",
+        sanity_restart=False,
+        deploy_generation=0,
+        external_cockroach=True,
+    ),
+    Materialized(
+        name="mz_new",
+        sanity_restart=False,
+        deploy_generation=1,
+        restart="on-failure",
```

Comment:
Why is this only for `mz_new`?

Reply:
mz_new requires a restart to come up in write-mode after the promotion. mz_old we do't want to restart then, otherwise they will both be running again.